File: Compute/Compute.java
public class Compute{
	public static void main(String[] args) {

		int[] b={1,0,0,1,0,0,1,0,1};

		int a=compute(943,741,293,741,b);

		System.out.println("\n\na="+a);


	}

	public static int compute(int n,int m,int k, int a, int[] b){
		a=1;
		for(int i=b.length-1; i>=0; i--){
			System.out.println("\n\nFor i= "+i+" and b[i]=" + b[i]);
			a=(a*a)%n;
			System.out.println("a="+a);

			if (b[i]==1){
				System.out.println("a= ("+a+ "*"+ m+")="+ (a*m)%n);
				a=(a*m)%n;
			}
		}
		return a;
		}
		 
}


		

File: Compute/compute.py
def compute(n,m,k, a, b):
	for i in range(len(b),0):
		a=(a*a)%n
		print("a=",a)
		if (b[i]==1):
			print("(",a, "*", m,")=", (a*m)%n)
			a=(a*m)%n
	return a


b=[1,0,0,1,0,0,1,0,1,1]

print(compute(943,741,587,1,b))
